### PR TITLE
socket: Align struct sa_familty to allow safe casting between all sockaddr_*

### PR DESCRIPTION
**Pull request by agkaminski**
DONE: RTOS-546

## Description

Fixes https://github.com/phoenix-rtos/phoenix-rtos-project/issues/781

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [x] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [ ] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: include/sys/socket.h [hunk 1]
```diff
@@ -76,19 +76,20 @@ extern "C" {
...
-
+	char sa_data[14]; /* actual size may be bigger */
 };
 
 
 struct sockaddr_storage {
 	sa_family_t ss_family;
-	char        ss_data[128-sizeof(sa_family_t)];
+	uint8_t _ss_pad[8 - sizeof(sa_family_t)];    /* Insert max padding by hand to force whole struct zeoring */
+	char ss_data[128 - sizeof(sa_family_t) - 6]; /* e.g. struct sockaddr_un needs some space */
```
[the remark this answers is not in the document]
Comment (agkaminski) [reply include/sys/socket.h:92]: DLC39-C recommendation: https://wiki.sei.cmu.edu/confluence/display/c/DCL39-C.+Avoid+information+leakage+when+passing+a+structure+across+a+trust+boundary?focusedCommentId=88028754#comment-88028754

I'll add more specific comment here then

File: include/sys/socket.h [hunk 2]
```diff
@@ -76,19 +76,20 @@ extern "C" {
 #endif
 
 
-typedef uint16_t sa_family_t;	// match lwIP size: u8 len + u8 family
+/* Match lwIP size: u8 len + u8 family. Align to allow safe sockaddr_* casting */
+typedef uint16_t sa_family_t __attribute__((aligned(8)));
```
[the remark this answers is not in the document]
Comment (agkaminski) [reply include/sys/socket.h:80]: Your test doesn't pass if alignment is added to the sockaddr_storage struct only

File: include/sys/socket.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (xvuko) [reply include/sys/socket.h]:
```
struct sockaddr_storage {
	sa_family_t ss_family;
	char ss_data[128 - sizeof(sa_family_t)]; 
};
```
Compiler will not add padding to this structure. `ss_data` is a padding member here.

File: include/sys/socket.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (xvuko) [reply include/sys/socket.h:80]: Indeed. This means my test needs to be changed :smile: 
```c
_Static_assert(offsetof(struct ast, addr) >= offsetof(struct ain6, addr), "misaligned");
```

File: include/sys/socket.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (agkaminski) [reply include/sys/socket.h]: Brain fart, padding only needed when using aligning struct member. Fixed

File: include/sys/socket.h [hunk 3]
```diff
@@ -76,46 +76,46 @@ extern "C" {
 #endif
 
 
-typedef uint16_t sa_family_t;	// match lwIP size: u8 len + u8 family
+/* Match lwIP size: u8 len + u8 family. Align to allow safe sockaddr_* casting */
```
Comment (gerard5) [inline include/sys/socket.h:79]: Once added, now removed [`__attribute__(aligned(4))`](https://github.com/phoenix-rtos/libphoenix/commit/41d44ee02ac230c4578d928f76fb094d55cf8fdd#diff-34ef7636426053b9b87fd53d7eeb2100d372858fd0b755a4191ce5bcb2efb191R79-R80). Comment not updated.
```suggestion
/* Match lwIP size: u8 len + u8 family. */
```

**Review by xvuko — APPROVED** [review]

**Pull request merged**